de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/ui/views/buttons/ConsistencyButton.java

```diff
@@ -68,14 +68,24 @@ public void actionPerformed(ActionEvent e) {
       new ISessionLifecycleListener() {
         @Override
         public void sessionStarted(ISarosSession newSarosSession) {
-          setSarosSession(newSarosSession);
-          setEnabledFromUIThread(true);
+          if (!newSarosSession.isHost()) {
+            setSarosSession(newSarosSession);
+            setEnabledFromUIThread(true);
+
+          } else {
+            setToolTipText("Recover inconsistencies - Disabled for host");
+          }
         }
 
         @Override
         public void sessionEnded(ISarosSession oldSarosSession, SessionEndReason reason) {
-          setSarosSession(null);
-          setEnabledFromUIThread(false);
+          if (!oldSarosSession.isHost()) {
+            setSarosSession(null);
+            setEnabledFromUIThread(false);
+
+          } else {
+            setToolTipText("Recover inconsistencies");
```

Comment:
I guess the tooltip should be the same when not in a session regardless if you were the session host or not in a former session.

Reply:
The tooltip never gets updated for the client, so this is only needed for the host. But yes, since updating the tooltip would be a good idea, always resetting it when the session ends is the way to go.